setup.py

```diff
@@ -102,11 +70,5 @@
     include_package_data=True,
     entry_points={'console_scripts': ['bokeh = bokeh.__main__:main']},
     zip_safe=False,
-    cmdclass=versioneer.get_cmdclass(),
+    cmdclass=versioneer.get_cmdclass({"build_ext": BuildJSCmd}),
```

Comment:
Does Bokeh have C extension code? If not, I expect that this will never get invoked –– setuptools does not invoke build_ext if there’s no C extension to compile.

Reply:
No C code, only a Typescript compilation

Reply:
I only did it this way because there did not seem to be any other way to get things working with both regular and editable installs, while also only changing one thing in one place.

Reply:
You can override the build_py command, with a subclass that calls `super().run()` and builds JS before that call in the overridden run method.

Reply:
Yeah, I haven't thought about that, but that seems to be the easiest approach so far.

Reply:
https://stackoverflow.com/questions/51032759/python-packaging-with-pth-files/71137826#71137826 has a code sample, with a similar approach for generating a different file.

Reply:
I did think about using `build_py` but I guess it seemed like an abuse, since it would be building something other than python.

Reply:
It's what I'd do. :)